File: 633-sum-of-square-numbers/sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        a, b = 0, math.floor(math.sqrt(c))
        while a <= b : 
            product = (a * a) + (b * b)
            if product == c : return True
            if product > c : b-=1
            else : a+=1
        return False

        # what do i do when if we go less we go too less?
        

File: 633-sum-of-square-numbers/sum-of-square-numbers.js
/**
 * @param {number} c
 * @return {boolean}
 */
var judgeSquareSum = function(c) {
    let start = 0, end = Math.ceil(Math.sqrt(c))
    while(start<=end){
        let res = (start*start) + (end * end)
        if (res == c)return true
        if(res > c)end--
        else start++
    }
    return false
};